### ISSUE TITLE
Able require external modules that have not been specified under require.external = [ ]

### ISSUE THREAD

**Issue by NishiBangar** [patriksimek/vm2#96]
I want to point out that, during vm2 implementation, i have come across a situation,  where i have some code (custom logic stored in a string variable) that needs to be executed in the sandbox environment.
*  Firstly, i have a helper script stored in a separate js file, **sandboxHelperScript.js**, this acts as wrapper in order to execute custom logic and also to require all the necessary custom and external modules.
```javascript
(function() { 
	 console.log("url");
         var url = require('url');
         console.log('async');
	 var async = require('async');
        // ****************************************************
        // ********** Custom External Module ***************
         var customModule= require(customModulesPath+'customModuleScript');
	 var customModuleOb j= new customModule();
        // ****************************************************
        // ****************************************************
         var responseObj = {};
  	 eval(script.userScript);   //   <------- Custom logic 
     return responseObj;
 })();
```

* Next, i am reading this helper file in main.js, in order run in the sandbox environment. 
```javascript
// **** Custom Logic  *****
 var userScript = customLogic;
//**** Read helper script ****
  var helper = fs.readFileSync(__dirname + '/../CustomModule/sandboxHelperScript.js', { encoding: "utf8" });
```

* vm2 configuration in main.js 
```javascript
            const {NodeVM, VMScript} = require('vm2');
            const vm = new NodeVM({
                console: 'inherit',
                sandbox: {
                    customModulesPath: __dirname + '/../CustomModule/',
                    request : requestObj,
                    script: {
                        userScript: userScript    // <------ Custom Logic
                    } 
                },
                require: {
                    context : 'sandbox',
                    external: ['async'],    // <-------- External module restriction 
                    builtin: ['url'],
                    root: './'
                    }
             });
   // **** Run custom logic in sandbox environment *****
       try{
           var result = vm.run("module.exports = "+helper,"main.js");
        }
        catch(e){
            console.log(e.name);
            console.log(e.message);
        }
```

## Problem
If you notice, in vm2 configuration under external option, i have only mentioned **async** to be required ( ie. restricting external module access to only "async"). But, in addition to this, if you observe **sandboxHelperScript** at the top, , i am requiring a custom external module, **customModuleScript** , which therefore has not been mentioned under the external module access array. At this point, i am expecting an error, but i am able to require this module with ease.
In other words, if "**require.external = false**", as expected, i am not able require any external modules, but on the contrary, if, "**require.external = ['','' ]**",then,  there is no external module require check happening for specified modules under this array.

I wanted things in a way, such that,  i should be restricting the external module access in sandbox environment to whatever the modules that has been specified. 

Kindly, let me know whether there is a particular way to handle things as i require.
Thanks in advance.
